### PR TITLE
remove quotes

### LINKED ISSUE #45: .eq doesn't work with querying foreign tables
# Bug report

## Describe the bug

.eq querying a foreign table doesn't work

## To Reproduce

Try to query a foreign table with .eq

## Expected behavior

This is the equivalent JavaScript query and it works just fine.

Screenshot 2022-11-18 at 1 53 29 PM

## Additional context

I have a table called `schedule` and a table called `schedule_details` that stores information on each individual day (so there are 7 in total). I am trying to reach the table `schedule_details` and do .eq(schedule_details.purchase_day, 5) where I only get the information about today.

Screenshot 2022-11-18 at 1 47 59 PM

This is the error that comes back

Screenshot 2022-11-18 at 1 52 10 PM

### PR DESCRIPTION
**Pull request by spencerbart**
Bug fix

#45 #42

### CODE REVIEW

**Comment by spencerbart** [comment]
Screenshot 2022-11-18 at 2 16 22 PM

These are the same queries. On the left is the JavaScript SDK. On the right is the Rust SDK. The only difference is that Rust inserts unnecessary quotes that don't allow for numbers or foreign table queries.

**Comment by spencerbart** [comment]
Screenshot 2022-11-18 at 2 22 42 PM
Screenshot 2022-11-18 at 2 22 35 PM

The top is JavaScript SDK. The bottom is Rust SDK. Same queries but JavaScript works whereas Rust does not.

**Comment by spencerbart** [comment]
Screenshot 2022-11-18 at 2 39 52 PM

Here's another example of a `.gt` query putting unnecessary quotes. On the left is JavaScript and on the right is Rust

File: src/filter.rs [hunk 1]
```diff
@@ -4,7 +4,7 @@ use crate::Builder;
 
 fn clean_param(param: &str) -> Cow<str> {
     if ",.:()".chars().any(|c| param.contains(c)) {
```
Comment (spencerbart) [inline src/filter.rs:6]: We could also just remove the "." from the clean_param check.

**Review by soedirgo — APPROVED** [review]

**Comment by soedirgo** [comment]
Thanks! Do you mind addressing the failing check as well?

**Comment by spencerbart** [comment]
@soedirgo I just fixed the clippy issue. I also realized that `clear_params` is now redundant so I removed it. I just checked through the JavaScript SDK (postgrest-js) and it appears that it doesn't have any function that clears the params. 

This is in [PostgrestFilterBuilder.ts](https://github.com/supabase/postgrest-js/blob/master/src/PostgrestFilterBuilder.ts)
Screenshot 2022-11-21 at 11 18 39 PM

**Pull request merged**